### PR TITLE
Added documentation for FGAC for Flow Framework

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
Added documentation for FGAC for Flow Framework. 
Reference PR https://github.com/opensearch-project/flow-framework/pull/838
Reference Issue https://github.com/opensearch-project/flow-framework/issues/668

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._

### Version
_List the OpenSearch version to which this PR applies, e.g. 2.14, 2.12--2.14, or all._

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _automating-configurations/security.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+Use backend roles to configure fine-grained access to individual workflows based on roles. For example, users of different departments in an organization can view workflows owned by their own department.
+
+First, make sure your users have the appropriate [backend roles]({{site.url}}{{site.baseurl}}/security/access-control/index/). Backend roles usually come from an [LDAP server]({{site.url}}{{site.baseurl}}/security/configuration/ldap/) or [SAML provider]({{site.url}}{{site.baseurl}}/security/configuration/saml/), but if you use the internal user database, you can use the REST API to [add them manually]({{site.url}}{{site.baseurl}}/security/access-control/api#create-user).
+
+Next, enable the following setting:
+
+```json
+PUT _cluster/settings
+{
+  "transient": {
```
Comment (dbwiddis) [inline _automating-configurations/security.md:39]: should this be transient or persistent?  This seems appropriate as an example but not sure we want it in the "how to" section without specifying why one or the other.

Reading further this all seems like an example and should have its own "sample" or "example" header.

File: _automating-configurations/security.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
+---
+layout: default
+title: Workflow security
+nav_order: 25
+---
+
+# Flow Framework security
```
[the remark this answers is not in the document]
Comment (owaiskazi19) [reply _automating-configurations/security.md]: Kept it as security to stay consistent with other plugins https://opensearch.org/docs/latest/observing-your-data/ad/security/. Let me know if you think otherwise

File: _automating-configurations/security.md (hunk 1 above)
Comment (owaiskazi19) [reply _automating-configurations/security.md:38]: This section explains the overall flow with the settings that needs be enabled.

Comment (dbwiddis) [reply _automating-configurations/security.md:38]: I’ll let the doc review address that.

**Review by dbwiddis — APPROVED** [review]

File: _automating-configurations/security.md [hunk 3]
```diff
@@ -4,31 +4,31 @@ title: Workflow security
 nav_order: 25
```
Comment (owaiskazi19) [inline _automating-configurations/security.md:4]: Does the navigation looks good @kolchfa-aws ?

File: _automating-configurations/security.md [hunk 4]
```diff
@@ -4,31 +4,31 @@ title: Workflow security
...
 ---
 
-# Flow Framework Security
+# Workflow security
```
Comment (owaiskazi19) [inline _automating-configurations/security.md:7]: @dbwiddis Should we keep it as Workflow security or Flow Framework?

File: _automating-configurations/security.md [hunk 5]
```diff
@@ -86,4 +88,6 @@ PUT _plugins/_security/api/rolesmapping/flow_framework_full_access
...
\ No newline at end of file
+Users without backend roles can still view other users' workflow results if they have `flow_framework_read_access`. This also applies to users who have `flow_framework_full_access`, because this permission includes all of the permissions of `flow_framework_read_access`. 
+
+Administrators should inform users that the flow_framework_read_access permission allows them to view the results of any workflow in the cluster, including data not directly accessible to them. To limit access to results of a specific workflow, administrators should apply backend role filters when creating the workflow. This ensures that only users with matching backend roles can access that workflow's results.
```
Comment (owaiskazi19) [inline _automating-configurations/security.md:93]: @kolchfa-aws should be `flow_framework_read_access`

File: _automating-configurations/api/index.md [hunk 6]
```diff
@@ -18,4 +18,6 @@ OpenSearch supports the following workflow APIs:
 * [Search workflow]({{site.url}}{{site.baseurl}}/automating-configurations/api/search-workflow/)
 * [Search workflow state]({{site.url}}{{site.baseurl}}/automating-configurations/api/search-workflow-state/)
 * [Deprovision workflow]({{site.url}}{{site.baseurl}}/automating-configurations/api/deprovision-workflow/)
-* [Delete workflow]({{site.url}}{{site.baseurl}}/automating-configurations/api/delete-workflow/)
\ No newline at end of file
+* [Delete workflow]({{site.url}}{{site.baseurl}}/automating-configurations/api/delete-workflow/)
+
+For information about workflow API security, see [Workflow security]({{site.url}}{{site.baseurl}}/automating-configurations/workflow-security/).
```
Comment (owaiskazi19) [inline _automating-configurations/api/index.md:23]: `For information about workflow access control, see...`

File: _automating-configurations/security.md (hunk 4 above)
Comment (dbwiddis) [reply _automating-configurations/security.md:7]: The word "workflow" is commonly used in other places and has led to a lot of confusion.  So other than the API and documentation directly relating to it, I'd like to avoid using "workflow".

That said, the left-column header for our plugin is "Automating Configurations" and it seems that may be appropriate as well, although fitting that together with "security" seems awkward.

TLDR: I trust doc writers to pick here. I'm against "Workflow security" and prefer "Flow Framework" if that's the only option, but am very open to other headers.

Comment (kolchfa-aws) [reply _automating-configurations/security.md:7]: I'm equally not sure about the best heading for this topic. I would go with a consistent title "Automating workflows", "Workflow APIs", "Workflow security", etc. But we're already using "Automating configurations" and then everything else mentions "workflows". Adding "Flow Framework" might make the user confused about what we're referencing. That said, I see your point about workflow being a generic concept so I reverted to "Flow Framework security" and added the first sentence that explains that workflows are provided by the Flow Framework plugin. Maybe @natebower can weigh in on this during editorial review.

File: _automating-configurations/api/index.md (hunk 6 above)
Comment (natebower) [inline _automating-configurations/api/index.md:23]: Do we mean "API workflow security"?

File: _automating-configurations/workflow-security.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
+---
+layout: default
+title: Flow Framework security
+nav_order: 50
+---
+
+# Flow Framework security
+
+In OpenSearch, automated workflow configurations are provided by the Flow Framework plugin. You can use the Security plugin together with the Flow Framework plugin in OpenSearch to limit non-admin users to specific actions. For example, you might want some users to only be able to create, update, or delete workflows, while others may only view workflows.
```
Comment (natebower) [inline _automating-configurations/workflow-security.md:9]:
```suggestion
In OpenSearch, automated workflow configurations are provided by the Flow Framework plugin. You can use the Security plugin together with the Flow Framework plugin to limit non-admin users to specific actions. For example, you might want some users to only be able to create, update, or delete workflows, while others may only be able to view workflows.
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @owaiskazi19 Please see my comments and changes and let me know if you have any questions. Thanks!

File: _automating-configurations/api/index.md (hunk 6 above)
Comment (kolchfa-aws) [inline _automating-configurations/api/index.md:23]:
```suggestion
For information about workflow access control, see [Workflow security]({{site.url}}{{site.baseurl}}/automating-configurations/workflow-security/).
```

File: _automating-configurations/index.md [hunk 8]
```diff
@@ -44,3 +44,4 @@ Workflow automation provides the following benefits:
 - For the workflow step syntax, see [Workflow steps]({{site.url}}{{site.baseurl}}/automating-configurations/workflow-steps/).  
 - For a complete example, see [Workflow tutorial]({{site.url}}{{site.baseurl}}/automating-configurations/workflow-tutorial/).
 - For configurable settings, see [Workflow settings]({{site.url}}{{site.baseurl}}/automating-configurations/workflow-settings/).
+- For information about workflow security, see [Workflow security]({{site.url}}{{site.baseurl}}/automating-configurations/workflow-security/).
```
Comment (kolchfa-aws) [inline _automating-configurations/index.md:47]:
```suggestion
- For information about workflow access control, see [Workflow security]({{site.url}}{{site.baseurl}}/automating-configurations/workflow-security/).
```

File: _automating-configurations/workflow-security.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+  }
+}
+```
+{% include copy-curl.html %}
+
+Now when users view workflow resources in OpenSearch Dashboards (or make REST API calls), they only see workflows created by users who share at least one backend role.
+
+For example, consider two users: `alice` and `bob`.
+
+`alice` has an analyst backend role:
```
Comment (kolchfa-aws) [inline _automating-configurations/workflow-security.md:49]:
```suggestion
`alice` has an `analyst` backend role:
```

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @owaiskazi19!

File: _automating-configurations/security.md (hunk 4 above)
Comment (kolchfa-aws) [reply _automating-configurations/security.md]: I like that! Let me change to "workflow template security"

File: _automating-configurations/security.md (hunk 3 above)
Comment (kolchfa-aws) [reply _automating-configurations/security.md]: I moved this topic down so it's the last in the left nav.

File: _automating-configurations/security.md (hunk 4 above)
Comment (kolchfa-aws) [reply _automating-configurations/security.md]: @owaiskazi19 @dbwiddis I changed the topic name to "Workflow template security".
